DMX.Core.Api.Tests.Unit/Services/Orchestrations/LabOrchestrationServiceTests.Exceptions.Add.cs

```diff
@@ -0,0 +1,109 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using DMX.Core.Api.Models.Foundations.Labs;
+using DMX.Core.Api.Models.Orchestrations.Labs.Exceptions;
+using FluentAssertions;
+using Force.DeepCloner;
+using Moq;
+using Xeptions;
+using Xunit;
+
+namespace DMX.Core.Api.Tests.Unit.Services.Orchestrations
+{
+    public partial class LabOrchestrationServiceTests
+    {
+        [Theory]
+        [MemberData(nameof(DependencyExceptions))]
+        public async Task ShouldThrowOrchestrationDependencyExceptionOnAddIfDependencyErrorOccursAndLogItAsync(
+            Xeption dependencyException)
+        {
+            // given
+            Lab randomLab = CreateRandomLab();
+            Lab inputLab = randomLab.DeepClone();
+
+            var expectedLabOrchestrationDependencyException =
+                new LabOrchestrationDependencyException(
+                    dependencyException.InnerException as Xeption);
+
+            this.labServiceMock.Setup(service =>
+                service.AddLabAsync(inputLab))
+                    .ThrowsAsync(dependencyException);
+
+            // when
+            ValueTask<Lab> addLabTask =
+                this.labOrchestrationService.AddLabAsync(inputLab);
+
+            LabOrchestrationDependencyException actualLabOrchestrationDependencyException =
+                await Assert.ThrowsAsync<LabOrchestrationDependencyException>(
+                    addLabTask.AsTask);
+
+            // then
+            actualLabOrchestrationDependencyException
+                .Should().BeEquivalentTo(
+                    expectedLabOrchestrationDependencyException);
+
+            this.labServiceMock.Verify(service =>
+                service.AddLabAsync(It.IsAny<Lab>()),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedLabOrchestrationDependencyException))),
+                        Times.Once);
+
+            this.labServiceMock.VerifyNoOtherCalls();
+            this.externalLabServiceMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();

```

Comment:
This should be in the proper chronological order ma bro

Reply:
Fixed throughout all test files.